tsdb/db.go

```diff
@@ -384,26 +384,37 @@ var ErrClosed = errors.New("db already closed")
 // Current implementation doesn't support concurrency so
 // all API calls should happen in the same go routine.
 type DBReadOnly struct {
-	logger  log.Logger
-	dir     string
-	closers []io.Closer
-	closed  chan struct{}
+	logger     log.Logger
+	dir        string
+	sandboxDir string
+	closers    []io.Closer
+	closed     chan struct{}
 }
 
 // OpenDBReadOnly opens DB in the given directory for read only operations.
-func OpenDBReadOnly(dir string, l log.Logger) (*DBReadOnly, error) {
+func OpenDBReadOnly(dir, sandboxDir string, l log.Logger) (*DBReadOnly, error) {
```

Comment:
maybe we just want to inject it after initialization? (to avoid the others from passing `""`)?

Reply:
A more important question, should we keep `sandboxDir` optional even though we know a RO Head can alter the chunks dir and push the DB to panic?

The RW DB will panic only if it's running at the same time though. Some users may only be running `OpenDBReadOnly` and I don't think we should force them to add this layer of protection (to set `sandboxDir` and/or make sure it has enough space).

(I suppose that the RW DB and `OpenDBReadOnly` are compatible: if `OpenDBReadOnly` creates/truncates chunks while a RW DB is not running, RW DB will not complain the next time it's run)